apps/gallery/js/ImageEditor.js

```diff
@@ -1221,8 +1229,51 @@ ImageEditor.prototype.getFullSizeBlob = function(type, done, progress) {
       setTimeout(function() {
         done(blob);
       });
+    }
+
+    function updateExif(blob, metaData, callback) {
+      metaData.Orientation = 1;
+      if (metaData.PixelXDimension) {
+        metaData.PixelXDimension = canvas.width;
+      }
+      if (metaData.PixelYDimension) {
+        metaData.PixelYDimension = canvas.height;
+      }
```

Comment:
Do you want to set DateTime to the current time and ensure that DateTimeOriginal is set to the original time here?  If we modify the app to sort photos by their EXIF date instead of their file date, that will be helpful to have.  But it doesn't have to be part of this patch.